### PR TITLE
added raw data paths, used os module for file paths and added main entrypoint

### PR DESCRIPTION
**Pull request by zsyed15**
Added condig file that uses raw data paths and replaced all hard coded file paths to use os modules to avoid issues w/ differences between operatng systsems. Also made label_prep.py runnable via the main function entry point.

Here are 2 examples on how it can be run: 
python -m risknet.run.pipeline

You can also run it by passing command line arguments to the main functions as shown below 
python src/risknet/main.py --job risknet.proc.label_prep --job-args fm_root="~/" data="historical_data_time_2009Q1.txt","dev_labels.pkl","dev_reg_labels.pkl"

### CODE REVIEW

**Comment by zsyed15** [comment]
Updated and tested the entrypoint/config file on DSMLP, and it seems to work fine. Files like pipeline.py can be run with the following command: ```python -m risknet.run.pipeline```. You can also run an individual module with an execute method (currently only added label_prep) using ```python src/risknet/main.py --job risknet.proc.label_prep```.

File: src/risknet/main.py [hunk 1]
```diff
@@ -66,10 +66,13 @@ def run_job(args: Namespace, job_args: Dict[str, str]) -> None:
...
 if __name__ == "__main__":
+    #Currently only the following modules can be run from main : label_prep.py,pipeline.py
+    #add an execute function to a module to make it runnable via main entrypoint. 
     args = get_args()
-
+    print(args)
     job_args, env = get_job_args(args)
-
+    
+    print('job_args',job_args)
```
Comment (rodalbuyeh) [inline src/risknet/main.py:75]: can we make this a logger.info statement instead of print

File: src/risknet/run/pipeline.py [hunk 2]
```diff
@@ -85,4 +95,7 @@
 
 print(auc)
 print(pr)
-print(recall)
\ No newline at end of file
+print(recall)
```
Comment (rodalbuyeh) [inline src/risknet/run/pipeline.py:98]: for all of these, make them logger.info statements

**Review by rodalbuyeh — CHANGES_REQUESTED** [review]
Please implement a logger instead of print statements

File: src/risknet/config/conf_template.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# COPY EVERYTHING BELOW INTO conf.yaml (Create if doesn't exist)
+general:
+  app_name: risknet
+  log_bucket: gs://albell-logs/
+  dask_cluster_type: local
+  gcp_dask_workers: 4
+data:
+  titanic_root: gs://albell-test/
+  fm_root : "YOUR_FM_ROOT" #TODO: Add your fm_root
```
Comment (emchin) [inline src/risknet/config/conf_template.yaml:9]: In the future, we plan for everyone to use DSMLP. Can we set `fm_root = "../../../../../teams/a15/data/"` now? Or (maybe better) can we add `fm_root = ../../../../../teams/a15/data/` as a comment here?

**Review by emchin — APPROVED** [review]
Everything looks good here! **I trust that you have a reason for creating a new config_template.yaml instead of appending to config.yaml?** 

Otherwise, just one comment -- adding the DSMLP fm_root in config_template.yaml might be convenient.

File: .gitignore [hunk 4]
```diff
@@ -66,6 +66,9 @@ local_settings.py
 db.sqlite3
 db.sqlite3-journal
 
+#config
+/src/risknet/config/conf.yaml
```
Comment (rodalbuyeh) [inline .gitignore:70]: why would you gitignore this? If you check the handler, it also accepts a path to an environment variable.

File: setup.cfg [hunk 5]
```diff
@@ -25,10 +25,10 @@ package_dir =
     =src
 include_package_data = True
 packages = find:
-python_requires = >= 3.10
+python_requires = >= 3.9.5
 install_requires =
     numpy==1.26.1
-    pandas==2.1.2
+    pandas[performance]==2.1.2
```
Comment (rodalbuyeh) [inline setup.cfg:31]: what's pandas performance?

File: src/risknet/config/conf_template.yaml (hunk 3 above)
Comment (rodalbuyeh) [reply src/risknet/config/conf_template.yaml:9]: yes, you can add it as a path if that's the working path. Alternatively you can hard code it in the config module's __init__.py or have a constants.py module somewhere.

File: src/risknet/main.py [hunk 6]
```diff
@@ -38,6 +37,7 @@ def get_job_args(arguments: Namespace) -> Tuple[Dict[str, str],
         job_args_tuples = [arg_str.split('=') for arg_str in
                            arguments.job_args]
         logger.info('job_args_tuples: %s' % job_args_tuples)
+        print(job_args_tuples)
```
Comment (rodalbuyeh) [inline src/risknet/main.py:40]: I'll go back on what I said earlier and indicate that we should have logging. We can do a separate PR for this though.

File: src/risknet/main.py (hunk 1 above)
Comment (rodalbuyeh) [reply src/risknet/main.py:75]: Note that the reason for logs is that they don't just get printed to the console, they can get timestamped and written to a .log file so they're persistent. Definitely useful for keeping track of the model lifecycle.

File: src/risknet/proc/label_prep.py [hunk 7]
```diff
@@ -68,4 +70,20 @@ def label_proc(fm_root, label_sets):
         regression_flagged_loans: DataFrame = pd.DataFrame(performance_df.loc[performance_df['default'] == 0].groupby("loan_sequence_number")['progress'].max()).reset_index().rename(columns={'progress':'undefaulted_progress'})
 
         with open(fm_root + i[2], 'wb') as f:
-            pickle.dump(regression_flagged_loans, f)
\ No newline at end of file
+            pickle.dump(regression_flagged_loans, f)
+            
+
+         
+def execute(fm_root=None,data=None):
```
Comment (rodalbuyeh) [inline src/risknet/proc/label_prep.py:77]: we should probably change these execute function names to 'main'

**Review by rodalbuyeh — APPROVED** [review]
Overall looks fine, I'd get that config.yaml situation worked out in a way that works for everybody (if possible). Otherwise, use the env variable handling in the handler.py instead of gitignoring the conf.yaml.

File: setup.cfg (hunk 5 above)
Comment (tambat22) [reply setup.cfg:31]: Pandas performance has dependencies like `bottleneck` and `numba` which are needed to run RiskNet on DSMLP. Also, according to the pandas docs, these dependencies "provide speed improvements, especially when working with large data sets".

**Pull request merged**